pageobjects/homepage.js

```diff
@@ -1,76 +1,106 @@
 var findButton = '//a[@role="button"]'
 var closeBedsfilter = '//button[@class="_85b30621"]'
-var showpropertytypes = '//*[@name="Category picker"]'
+var showPropertytypes = '//*[@name="Category picker"]'
+var selectedLocation = '//*[@aria-label = "Active filter label"]'
 //var selectedproperty
 
 //Page Functions
 var homepage = Object.create(page, { 
 
     salebtn : {get: function() {return $(saleButton)}},
 
-    enterlocation : {get: function() {homepageassertions.entertext1}},
+    enterlocation : {value: function(setLocation) {
+        $(locationBar).setValue(setLocation)
+        browser.pause(2000);
+        $(locationIndex).click();
+        console.log(setLocation + " is entered in the location box")
+    }},
     
     clickmoreoption :{get: function() {
-        var expectedoption = 'More Options'
-        if(browser.getText(moreFilteroptions) == expectedoption)
+        var expectedOption = 'More Options'
+        if(browser.getText(moreFilteroptions) == expectedOption)
         {
             return $(moreOption).click()
         }
         else{
             console.log("More Filters are already viewable")
         }
     }},
-    
-    selectpropertytype : {value: function(selectedproperty){
+
+    selectpropertytype : {value: function(selectedProperty){
         $(propertyTypebox).click();
         //var selectedproperty = 'Townhouse'
-        var propertytypelist = $(showpropertytypes);
-        var getPropertytypenames = $(showpropertytypes).getText('li');
-        var numberofproperties = getPropertytypenames.length
+        var propertyTypelist = $(showPropertytypes);
+        var getPropertytypenames = $(showPropertytypes).getText('li');
+        var numberofProperties = getPropertytypenames.length
 
-        for (var i =0 ; i<=numberofproperties ; i++)
+        for (var i =0 ; i<=numberofProperties ; i++)
         {
-            var propertyname = getPropertytypenames[i];
-            if(propertyname == selectedproperty)
+            var propertyName = getPropertytypenames[i];
+            if(propertyName == selectedProperty)
             {
-                propertytypelist.$$('li')[i].click();
+                propertyTypelist.$$('li')[i].click();
             }
         }    
     }},
 
-    selectprice : {get: function(){
+    selectMinprice : {value: function(minPrice){
         $(priceFilter).click();
-        $(minPricerange).click();
-        console.log($(minPricerange).getText() + ' is selected as minimum price')
+
+        var totalMinprices = $$(minPricerange).length;
+        for (var i = 0 ; i < totalMinprices ; i++ )
+        {
+            //var price = $(minPricerange)[i].getText()
+            if ($$(minPricerange)[i].getText() == minPrice)
+            {
+                $$(minPricerange)[i].click()
+                console.log(minPrice + ' is selected as the min price');
+            }
+        }
+    }},
+
+    selectMaxprice : {value: function(maxPrice){
```

Comment:
`selectMaxprice` and `selectMinprice` methods are very similar. Can you make a single method for both cases?

Reply:
Single Method used for both Min price and Max price